src/cmd/services/m3coordinator/ingest/write.go

```diff
@@ -58,11 +58,18 @@ type DownsamplerAndWriter interface {
 	WriteBatch(
 		ctx context.Context,
 		iter DownsampleAndWriteIter,
-	) error
+	) BatchError
 
 	Storage() storage.Storage
 }
 
+// BatchError allows for access to individual errors.
+type BatchError interface {
+	error
```

Comment:
Considering we’re only returning the last error in this path, can we add a new implementation of multierr which doesn’t accumulate?

Reply:
So I need to actually look at each error and determine 100% of them are bad request like errors, otherwise if there was one real error, the request needs to be retried.

I use `Errors()` to first establish all real errors are bad request like errors, then I use `LastError()` just to grab the one for logging purposes.  I could probably remote `LastError()` tbh from this interface.